app/models/piece.rb

```diff
@@ -153,7 +153,24 @@ def is_obstructed?(x1, y1, x2, y2)
         return true
       end
     end
-  end 
+  end
+
+  def valid_move?(new_x, new_y)
+    if square_occupied?(new_x, new_y) && self.color === game.get_piece_at_coor(new_x, new_y).color
+        return false
+    elsif
+      self.is_obstructed?(new_x, new_y)
+      return false
+    elsif
+      new_x === self.x_position && new_y === self.y_position
+      return false
+    else
+      off_board?(new_x new_y)
+      return false
+    end
+    return true
```

Comment:
Remember the refactoring points we talked about? we can get rid of all the if/else here and just have


def xyz
   some_method && some_other_check && some_other_check 
end

Reply:
I'll work on that tonight, thanks